kong/plugins/rate-limiting/handler.lua

```diff
@@ -64,6 +68,30 @@ local function get_usage(conf, api_id, identifier, current_timestamp, limits)
   return usage, stop
 end
 
+local function get_period_end_in_secs(current_timestamp, limits)
+  local timetable
+  local periods = timestamp.get_timestamps(current_timestamp)
+
+  for period, period_date in pairs(periods) do
+    if limits[period] then
+      timetable = new_from_timestamp(floor(period_date / 1000))
+
+      -- Unfortunately second & minute don't map correctly
+      if period == "second" then
+        timetable.sec = timetable.sec + 1
```

Comment:
Also, we have a new convention for `if/elseif/else` constructs; we add a blank line above `elseif` and `else`. (Please take a quick look at [this](https://github.com/Mashape/kong/blob/master/CONTRIBUTING.md#conditional-expressions).)